File: Backend/parent-service/agency-service/src/main/java/com/newsforyou/agencyservice/service/AgencyService.java
package com.newsforyou.agencyservice.service;

import com.newsforyou.agencyservice.dto.AgencyRequest;
import com.newsforyou.agencyservice.dto.AgencyResponse;
import com.newsforyou.agencyservice.dto.AgencyResponseList;
import com.newsforyou.agencyservice.model.Agency;

public interface AgencyService {
	
	public void createAgency(AgencyRequest agencyRequest);

	public AgencyResponseList getAllAgency();

	public AgencyResponse getAgency(String agencyId);

	public boolean checkAgencyAvailable(String agencyId);

	public void updateAgency(Agency agencyRequest);
}

File: Frontend/news-for-you/src/js/sevices/agencyService.js
import axios from "../../config/axiosConfig";
import { agencyFeedUrl, agencyUrl } from "../../config/constants";

const AgencyService = {
  getAgency: (agencyId) => {
    return axios.get(agencyUrl + "/" + agencyId);
  },
  getAllAgency: () => {
    return axios.get(agencyUrl + "/all-agency");
  },
  createAgency: (payload) => {
    return axios.post(agencyUrl, payload);
  },
  updateAgency: (payload) => {
    return axios.put(agencyUrl, payload);
  },
  getAllAgencyFeed: () => {
    return axios.get(agencyFeedUrl + "/all-agency-feed");
  },
  createAgencyFeed: (payload) => {
    return axios.post(agencyFeedUrl, payload);
  },
  getAgencyFeed: (id) => {
    return axios.get(agencyFeedUrl + "/" + id);
  },
};

export default AgencyService;
